File: app/src/main/java/com/rd/dotpagerview/utils/DensityUtils.java
package com.rd.dotpagerview.utils;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.DisplayMetrics;

public class DensityUtils {

    @SuppressWarnings("UnnecessaryLocalVariable")
    public static int dpToPx(@NonNull Context context, int dp) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        int px = Math.round(dp * (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT));

        return px;
    }

    @SuppressWarnings("UnnecessaryLocalVariable")
    public static int pxToDp(@NonNull Context context, int px) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        int dp = Math.round(px / (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT));

        return dp;
    }
}

File: pageindicatorview/src/main/java/com/rd/utils/DensityUtils.kt
package com.rd.utils

import android.content.res.Resources
import android.util.TypedValue

object DensityUtils {

	fun dpToPx(dp: Int): Int {
		return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp.toFloat(), Resources.getSystem().displayMetrics).toInt()
	}

	fun pxToDp(px: Float): Int {
		return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_PX, px, Resources.getSystem().displayMetrics).toInt()
	}
}
